text/0000-project-descriptor.md

```diff
@@ -0,0 +1,214 @@
+```
+
+The following sections describe each part of the schema in detail.
+
+## `[project]`
+
+The top-level `[project]` table may contain configuration about the repository, including `id` and `version`, but also metadata about how it is authored, documented, and version controlled.
+
+The `project.id`
+
+```toml
+[project]
+id = "<string>"
+name = "<string>"
+version = "<string>"
+```
+
+* `id` - the machine readable identifier of the project (ex. "com.example.myservice")
+* `name` - the human readable name of the project (ex. "My Example Service")
+* `version` - and arbitrary string representing the version of the project
+
+## `[project.licenses]`
+
+* `type` - This may use the [SPDX 2.1 license expression](https://spdx.org/spdx-specification-21-web-version), but is not limited to identifiers in the [SPDX Licenses List](https://spdx.org/licenses/).
+* `uri` - If this project is using a nonstandard license, then this key may be specified in lieu of or in addition to `type` to point to the license.
+
+## `[build.include]` and `[build.exclude]`
+
+A list of files to include in the build (while excluding everything else):
+
+```toml
+[build]
+include = [
+    "cmd/",
+    "go.mod",
+    "go.sum",
+    "*.go"
+]
+```
+
+A list of files to exclude from the build (while including everything else)
+
+```toml
+[build]
+exclude = [
+    "spec/"
+]
+```
+
+The `.gitignore` pattern is used in both cases. The `exclude` and `include` keys are mutually exclusive, and if both are present the Lifecycle will error out. These lists apply to both buildpacks built with `pack create-package` and apps built with `pack build`.
+
+Any files that are excluded (either via `include` or `exclude`) will be excluded before the build (i.e. not only exluded from the final image).
+
+If both `exclude` and `include` are defined, the build process will error out.
+
+## `[[build.buildpacks]]`
+
+The build table may contain an array of buildpacks. The schema for this table is:
+
+```toml
+[[build.buildpacks]]
+id = "<buildpack ID (optional)>"
+version = "<buildpack version (optional default=latest)>"
+uri = "<url or path to the buildpack (optional default=urn:buildpack:<id>)"
+```
+
+This defines the buildpacks that a platform should use on the repo.
+
+Either an `id` or a `uri` are required, but not both. If `uri` is provided, `version` is not allowed.
+
+## `[[build.env]]`
+
+Used to set environment variables at build time, for example:
+
+```toml
+[[build.env]]
+name = "JAVA_OPTS"
+value = "-Xmx1g"
+```
+
+## `[metadata]`
+
+This table includes a some defined keys, but additional keys are not validated. It can be used to add platform specific metadata. For example:
+
+```toml
+[metadata.heroku]
+pipeline = "foobar"
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The `project.toml` contents will be read by the platform and/or the lifecycle. The launch image will contain labels for the data in all fields except `include` and `exclude`.
```

Comment:
Should all of `[build]` be excluded from being added to the label? I don't see the value in replicating the buildpacks which are already stored in another label. In general, I would think that we would only add informational data to the label not so much configuration.

Reply:
agree, we should omit `build`